### PR TITLE
v6.0.2 (Mirrors, Exchange Reporting, and Bugs)

### LINKED ISSUE #575: Seed discovery returns incompatible and offline nodes
### Package Versions

Replace the values below using the output from `npm list storj`. Use ```npm list -g storj``` if installed globally.

```
GUI
Storj Share 4.0.1
Core Library 6.0.1
Protocol 1.0.0
```

Replace the values below using the output from `node --version`.

```
none
```

### Expected Behavior

Please describe the program's expected behavior. Include an example of your
usage code in the back ticks below if applicable.

Requesting seed list from server025.storj.dk
{"title":"Storj Bridge","version":"3.0.0","description":"Access the Storj network using a simple REST API. Hosted by Th3Van","x-protocol-version":"1.0.0","x-core-version":"6.0.1"}

I would expect a list of compatible nodes. No old protocol version and no offline nodes.

### Actual Behavior

Please describe the program's actual behavior. Please include any stack traces
or log output in the back ticks below.

```
{debug} [Sat Dec 03 2016 00:29:33 GMT+0100 (Mitteleuropäische Zeit)] Request: {"baseUrl":"http://server025.storj.dk","uri":"/contacts","method":"GET","timeout":30000,"qs":{"connected":true,"__nonce":"dad34303-aab5-43e4-a76a-41788563f7d6"},"json":true}
{debug} [Sat Dec 03 2016 00:29:36 GMT+0100 (Mitteleuropäische Zeit)] Body: [
{"lastSeen":"2016-12-02T23:29:32.123Z","port":4002,"address":"77.70.66.52","userAgent":"5.1.2","protocol":"0.10.0","nodeID":"87d9746cc17b6a4f8ab9c6b8b0e77e189a2452da"},
{"protocol":"0.10.0","userAgent":"5.1.2","address":"138.201.17.121","port":4153,"lastSeen":"2016-12-02T23:29:32.120Z","nodeID":"8faa224a4c37dc971c11946ed1f1614acd1a3dff"},
{"protocol":"0.10.0","userAgent":"5.1.0","address":"104.196.196.231","port":8472,"lastSeen":"2016-12-02T23:29:32.112Z","nodeID":"22901935a4a3eebdc1c462cb7448d667201aa8ad"},
{"protocol":"0.10.0","userAgent":"5.1.0","address":"104.196.39.49","port":8448,"lastSeen":"2016-12-02T23:29:32.112Z","nodeID":"0434996c6148537357bb8127018f88ea94c8d0cf"},
{"lastSeen":"2016-12-02T23:29:32.106Z","port":26399,"address":"24.90.108.93","userAgent":"5.0.1","protocol":"0.10.0","nodeID":"876bffe6d2117139ba1fe98702281da355473adc"},
{"lastSeen":"2016-12-02T23:29:32.102Z","port":44427,"address":"108.63.248.115","userAgent":"5.0.1","protocol":"0.10.0","nodeID":"a9c832465a074654867e3076f660de33e7f1cb37"},
{"protocol":"0.10.0","userAgent":"5.0.1","address":"89.164.183.136","port":53395,"lastSeen":"2016-12-02T23:29:32.094Z","nodeID":"e92a8bf719e3a885b4ad11166b33b8dd9f136cf1"},
...
```

### Steps to Reproduce

Please include the steps the reproduce the issue, numbered below. Include as
much detail as possible.

1. ...
2. ...
3. ...

### LINKED ISSUE #577: Uncaught TypeError: res.status(...).end is not a function
### Package Versions

Replace the values below using the output from `npm list storj`. Use ```npm list -g storj``` if installed globally.

```
GUI
Storj Share 4.0.1
Core Library 6.0.1
Protocol 1.0.0
```

Replace the values below using the output from `node --version`.

```
none
```

### Expected Behavior

Please describe the program's expected behavior. Include an example of your
usage code in the back ticks below if applicable.

No exception

### Actual Behavior

Please describe the program's actual behavior. Please include any stack traces
or log output in the back ticks below.

Uncaught TypeError: res.status(...).end is not a function
C:\Program Files\Storj Share\resources\app\node_modules\storj-lib\lib\network\transport.js:180
https://github.com/Storj/core/blob/master/lib/network/transport.js#L180

### Steps to Reproduce

Please include the steps the reproduce the issue, numbered below. Include as
much detail as possible.

1. ...
2. ...
3. ...

### PR DESCRIPTION
**Pull request by bookchin**
* Sends exchange report following a MIRROR operation
* Add `BridgeClient#listMirrorsForFile` (https://github.com/Storj/bridge/pull/300)
* Closes #577 
* Closes #575 
* Closes https://github.com/Storj/storjshare-cli/issues/226

### CODE REVIEW

**Comment by braydonf** [comment]
Testing with https://github.com/braydonf/storj-integration/tree/mirror and MIRROR_SUCCESS reports are working.

File: lib/bridge-client/index.js [hunk 1]
```diff
@@ -398,6 +398,21 @@ BridgeClient.prototype.replicateFileFromBucket = function(id, file, n, cb) {
 };
 
 /**
+ * Instructs the bridge to find N mirroring farmers for redundancy
```
Comment (braydonf) [inline lib/bridge-client/index.js]: jsdoc needs update

**Comment by braydonf** [comment]
LGTM

**Comment by CJClaude** [comment]
StorjShare: 9.0.6
Core:       6.0.9
Protocol:   1.0.0

assert.js:85
  throw new assert.AssertionError({
  ^
AssertionError: Invalid nodeID was supplied
    at Contact (/usr/lib64/node_modules/storjshare-cli/node_modules/kad/lib/contact.js:27:3)
    at Contact.AddressPortContact (/usr/lib64/node_modules/storjshare-cli/node_modules/kad/lib/contacts/address-port-contact.js:28:11)
    at new Contact (/usr/lib64/node_modules/storjshare-cli/node_modules/storj-lib/lib/network/contact.js:42:35)
    at FarmerInterface.Network._createContact (/usr/lib64/node_modules/storjshare-cli/node_modules/storj-lib/lib/network/index.js:382:10)
    at FarmerInterface.Network.connect (/usr/lib64/node_modules/storjshare-cli/node_modules/storj-lib/lib/network/index.js:371:33)
    at /usr/lib64/node_modules/storjshare-cli/node_modules/storj-lib/lib/network/index.js:268:12
    at wrappedIteratee (/usr/lib64/node_modules/storjshare-cli/node_modules/storj-lib/node_modules/async/dist/async.js:2712:13)
    at replenish (/usr/lib64/node_modules/storjshare-cli/node_modules/storj-lib/node_modules/async/dist/async.js:882:17)
    at iterateeCallback (/usr/lib64/node_modules/storjshare-cli/node_modules/storj-lib/node_modules/async/dist/async.js:867:17)
    at /usr/lib64/node_modules/storjshare-cli/node_modules/storj-lib/node_modules/async/dist/async.js:840:16
    at /usr/lib64/node_modules/storjshare-cli/node_modules/storj-lib/node_modules/async/dist/async.js:2720:21
    at /usr/lib64/node_modules/storjshare-cli/node_modules/storj-lib/lib/network/index.js:271:11
    at /usr/lib64/node_modules/storjshare-cli/node_modules/kad/lib/node.js:128:14
    at /usr/lib64/node_modules/storjshare-cli/node_modules/kad/node_modules/async/lib/async.js:541:30
    at /usr/lib64/node_modules/storjshare-cli/node_modules/kad/lib/router.js:539:14
    at /usr/lib64/node_modules/storjshare-cli/node_modules/kad/lib/router.js:234:14

**Pull request merged**